### ISSUE TITLE
`CUSTOM_GETTERS_SETTERS` false positive when a property has a backing field

### ISSUE THREAD

**Issue by 0x6675636b796f75676974687562** [saveourtool/diktat#1709]
## Describe the bug

Consider this idiomatic example [taken directly](https://kotlinlang.org/docs/properties.html#backing-properties) from [https://kotlinlang.org](kotlinlang.org):

```kotlin
package com.example

class MutableTableContainer {
    private var _table: Map<String, Int>? = null

    val table: Map<String, Int>
        get() {
            if (_table == null) {
                _table = hashMapOf()
            }
            return _table ?: throw AssertionError("Set to null by another thread")
        }
}
```

Diktat **1.2.5** warns:

```
MutableTableContainer.kt:7:9: [CUSTOM_GETTERS_SETTERS] custom getters and setters are not recommended, use class methods instead: get (cannot be auto-corrected) (diktat-ruleset:custom-getter-setter)
```

Probably we should make an exception for property accessors/modifiers whenever there's a backing field with the corresponding name (`xyz` &rarr; `_xyz`) and of the same type.

Related:

 - [x] #1711

## Expected behavior

No warning should have been reported.

## Observed behavior

A warning was reported.

## Steps to Reproduce

Check the above code example with _Diktat_.

## Environment information
* diktat version: **1.2.5**
* build tool (maven/gradle): any
* how is diktat run (CLI, plugin, etc.): doesn't matter
* kotlin version: **1.9**
* operating system: any
* link to a project (if your project is public):
